src/Widgets/EventRow.vala

```diff
@@ -70,6 +71,32 @@ public class DateTime.EventRow : Gtk.ListBoxRow {
         grid_context.add_class ("event");
         grid_context.add_provider (css_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
 
+        /* Color menuitem per calendar source of event */
+        get_style_calendar_color (cal_event.cal, grid);
+        get_style_calendar_color (cal_event.cal, event_image);
+
+        cal_event.cal.notify["color"].connect (() => {
+            get_style_calendar_color (cal_event.cal, grid);
+            get_style_calendar_color (cal_event.cal, event_image);
+        });
+
         add (grid);
     }
+
+    public void get_style_calendar_color (E.SourceCalendar cal, Gtk.Widget widget) {
```

Comment:
Since we're going to want this later for the event dots, we should just put this in Util and save ourselves that work in the other branch of moving it